pandas/tests/indexes/period/test_period.py

```diff
@@ -11,6 +11,36 @@
 from ..datetimelike import DatetimeLike
 
 
+class TestPeriodLevelMultiIndex(object):
+    # TODO: Is there a more appropriate place for these?
+    def test_set_index(self):
+        # GH#17112
+        index = Index(['PCE'] * 4, name='Variable')
+        data = [Period('2018Q2'),
+                Period('2021', freq='5A-Dec'),
+                Period('2026', freq='10A-Dec'),
+                Period('2017Q2')]
+        ser = Series(data, index=index, name='Period')
+        df = ser.to_frame()
+
+        res = df.set_index('Period', append=True)
```

Comment:
this could go in tests/frame where we do .set_index tests

don’t add commentary